internal/oonirun/v2.go

```diff
@@ -289,3 +301,38 @@ func v2MeasureHTTPS(ctx context.Context, config *LinkConfig, URL string) error {
 	// note: this function gracefully handles nil values
 	return V2MeasureDescriptor(ctx, config, newValue)
 }
+
+func maybeGetAuthenticationTokenFromFile(path string) (string, error) {
+	if path != "" {
+		return readFirstLineFromFile(path)
+	}
+	return "", nil
+}
+
+// readFirstLineFromFile reads the first line of the passed text file and trims any non-printable characters.
+func readFirstLineFromFile(filePath string) (string, error) {
+	f, err := os.Open(filePath)
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+
+	// Scan the first line
+	if scanner.Scan() {
+		line := scanner.Text()
+		trimmed := strings.TrimFunc(line, func(r rune) bool {
+			return !unicode.IsPrint(r)
```

Comment:
my intent was to strip any control chars like CR LF etc
but you're right that I could further try to tokenize and get only a valid b64 token

Reply:
fwiw, I started with the idea of adding arbitrary headers but then I decided to restrict the possibilities to a bearer token - which I think it will be enough for a simple case. 

I've changed the tokenizing function to be more strict and return only a valid b64 token, if found. please have a look at the new documentation and test cases!